spec/models/user_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User do
+  # rubocop:disable RSpec/ExampleLength
+  it 'populates a User object' do
+    # rubocop:disable Layout/LineLength
+    dataverse_user_group = 'cn=edu:berkeley:app:calnet-spa:group-spa-ucblibdataverse,ou=campus groups,dc=berkeley,dc=edu'
+    # rubocop:enable Layout/LineLength
+
+    auth = {
+      'provider' => 'calnet',
+      'extra' => {
+        'berkeleyEduAffiliations' => 'expected affiliation',
+        'displayName' => 'expected display name',
+        'berkeleyEduAlternateID' => 'expected email',
+        'uid' => 'expected UID'
+      }
+    }
+
+    [true, false].each do |dataverse_user|
```

Comment:
can you explain what this is doing? is the goal to check to see if it populates  the group membership? if that's the case, this might be better off being two discrete examples - one which does, and one which does not

Reply:
Done - and yes, much clearer now. The [true,false]... was a suggestion from CoPilot to make the spec shorter.